lua/ttt2/libraries/database.lua

```diff
@@ -0,0 +1,1419 @@
+	end
+
+	ValueReceived(request.index, request.itemName, request.key)
+	functionCache[identifier](isSuccess, value)
+end
+
+---
+-- Send query for setting a value to server
+-- @param table data contains index, itemName, key and value here
+-- @realm client
+-- @internal
+clientSendFunctions[MESSAGE_SET_VALUE] = function(data)
+	net.WriteUInt(data.index, uIntBits)
+
+	net.WriteString(data.itemName)
+	net.WriteString(data.key)
+	net.WriteString(tostring(data.value))
+end
+
+---
+-- Receive query for setting a value from client
+-- and set that value in the database if player is a superadmin
+-- @param string plyID64 the playerID64 of the player who sent the message
+-- @realm server
+-- @internal
+serverReceiveFunctions[MESSAGE_SET_VALUE] = function(plyID64)
+	local index = net.ReadUInt(uIntBits)
+	local itemName = net.ReadString()
+	local key = net.ReadString()
+	local value = net.ReadString()
+
+	local accessName = registeredDatabases[index].accessName
+	value = database.ConvertValueWithKey(value, accessName, key)
+
+	database.SetValue(accessName, itemName, key, value, plyID64)
+end
+
+---
+-- Send set value to client
+-- @param table data contains index, itemName, key, value here
+-- @realm server
+-- @internal
+serverSendFunctions[MESSAGE_SET_VALUE] = function(data)
+	net.WriteUInt(data.index, uIntBits)
+	net.WriteString(data.itemName)
+	net.WriteString(data.key)
+	net.WriteString(tostring(data.value))
+end
+
+clientReceiveFunctions[MESSAGE_SET_VALUE] = function()
+	local index = net.ReadUInt(uIntBits)
+	local itemName = net.ReadString()
+	local key = net.ReadString()
+	local value = net.ReadString()
+
+	value = database.ConvertValueWithKey(value, registeredDatabases[index].accessName, key)
+	OnChange(index, itemName, key, value)
+	ValueReceived(index, itemName, key)
+end
+
+---
+-- Send query for resetting the database to server
+-- @param table data contains index
+-- @realm client
+-- @internal
+clientSendFunctions[MESSAGE_RESET] = function(data)
+	net.WriteUInt(data.index, uIntBits)
+end
+
+---
+-- Receive query for resetting the database from client
+-- @param string plyID64 the playerID64 of the player who sent the message
+-- @realm server
+-- @internal
+serverReceiveFunctions[MESSAGE_RESET] = function(plyID64)
+	database.Reset(registeredDatabases[net.ReadUInt(uIntBits)].accessName, plyID64)
+end
+
+---
+-- Send reset command to client
+-- To make sure the client resets all stored values to the default
+-- @param table data contains index
+-- @realm server
+-- @internal
+serverSendFunctions[MESSAGE_RESET] = function(data)
+	net.WriteUInt(data.index, uIntBits)
+end
+
+---
+-- Receive reset command from server
+-- Reset all stored values back to the defaults
+-- @realm client
+-- @internal
+clientReceiveFunctions[MESSAGE_RESET] = function()
+	ResetDatabase(net.ReadUInt(uIntBits))
+end
+
+---
+-- Send query for getting a default value to server
+-- @warning This function is not yet implemented
```

Comment:
Well yeah I wanted to, because the code itself is not used anywhere. With the synced default values, this function is most likely never reached. So I only left a warning for edge cases that I forgot to think about and my code just keeps working normally.

Reply:
id rather remove them

Reply:
Can do that ^^